syncstorage-mysql/src/batch.rs

```diff
@@ -74,7 +83,7 @@ pub fn validate(db: &MysqlDb, params: params::ValidateBatch) -> DbResult<bool> {
         .filter(batch_uploads::batch_id.eq(&batch_id))
         .filter(batch_uploads::user_id.eq(&user_id))
         .filter(batch_uploads::collection_id.eq(&collection_id))
-        .get_result::<i32>(&db.conn)
+        .get_result::<i32>(&mut *db.conn.write()?)
```

Comment:
I think I recall you mentioned this is the part that acquires the write lock before inserting and we'll be able to change once we land some of the other patches, namely `&mut` for connection references, right?